TST Occasional failures of test_keyboard_interrupt

I have seen this test fail at least once in the CI and once locally. Rerunning the test fixes it, but it does suggest that it's maybe not very stable.
```
___________________________________________________________________________________ test_keyboard_interrupt[firefox] ____________________________________________________________________________________

selenium = <conftest.FirefoxWrapper object at 0x7f6963c4f7f0>

    def test_keyboard_interrupt(selenium):
>       assert selenium.run_js(
            """
            x = new Int8Array(1)
            pyodide._module.setInterruptBuffer(x)
            window.triggerKeyboardInterrupt = function(){
                x[0] = 2;
            }
            try {
                pyodide.runPython(`
                    from js import triggerKeyboardInterrupt
                    x = 0
                    while True:
                        x += 1
                        if x == 2000:
                            triggerKeyboardInterrupt()
                `)
            } catch(e){}
            return pyodide.runPython(`
                2000 < x < 2500
            `)
            """
        )
E       AssertionError: assert None
E        +  where None = <bound method SeleniumWrapper.run_js of <conftest.FirefoxWrapper object at 0x7f6963c4f7f0>>('\n        x = new Int8Array(1)\n        pyodide._module.setInterruptBuffer(x)\n        window.triggerKeyboardInterrup...       `)\n        } catch(e){}\n        return pyodide.runPython(`\n            2000 < x < 2500\n        `)\n        ')
E        +    where <bound method SeleniumWrapper.run_js of <conftest.FirefoxWrapper object at 0x7f6963c4f7f0>> = <conftest.FirefoxWrapper object at 0x7f6963c4f7f0>.run_js

src/tests/test_pyodide.py:198: AssertionError
```

On chrome it produces,
```
E       AssertionError: assert {}
E        +  where {} = <bound method SeleniumWrapper.run_js of <conftest.ChromeWrapper object at 0x7fa3564e7e80>>('\n        x = new Int8Array(1)\n        pyodide._module.setInterruptBuffer(x)\n        window.triggerKeyboardInterrup...       `)\n        } catch(e){}\n        return pyodide.runPython(`\n            2000 < x < 2500\n        `)\n        ')
E        +    where <bound method SeleniumWrapper.run_js of <conftest.ChromeWrapper object at 0x7fa3564e7e80>> = <conftest.ChromeWrapper object at 0x7fa3564e7e80>.run_js

src/tests/test_pyodide.py:198: AssertionError
--------------------------------------------------------------------------------------- Captured stdout teardown ----------------------------------------------------------------------------------------
Python exception:
Traceback (most recent call last):
  File "<exec>", line 6, in eval_code
  File "/lib/python3.8/site-packages/pyodide/_base.py", line 302, in eval_code
    return CodeRunner(
  File "/lib/python3.8/site-packages/pyodide/_base.py", line 225, in run
    exec(mod, self.globals, self.locals)
  File "<exec>", line 5, in <module>
KeyboardInterrupt
```

cc @hoodmane

--- a/src/tests/test_pyodide.py
+++ b/src/tests/test_pyodide.py
@@ -134,24 +134,25 @@ def test_eval_code_locals():
 
 
 def test_monkeypatch_eval_code(selenium):
-    selenium.run(
-        """
-        import pyodide
-        old_eval_code = pyodide.eval_code
-        x = 3
-        def eval_code(code, ns):
-            return [ns["x"], old_eval_code(code, ns)]
-        pyodide.eval_code = eval_code
-        """
-    )
-    assert selenium.run("x = 99; 5") == [3, 5]
-    assert selenium.run("7") == [99, 7]
-    selenium.run(
-        """
-        pyodide.eval_code = old_eval_code
-        """
-    )
-
+    try:
+        selenium.run(
+            """
+            import pyodide
+            old_eval_code = pyodide.eval_code
+            x = 3
+            def eval_code(code, ns):
+                return [ns["x"], old_eval_code(code, ns)]
+            pyodide.eval_code = eval_code
+            """
+        )
+        assert selenium.run("x = 99; 5") == [3, 5]
+        assert selenium.run("7") == [99, 7]
+    finally:
+        selenium.run(
+            """
+            pyodide.eval_code = old_eval_code
+            """
+        )
 
 def test_hiwire_is_promise(selenium):
     for s in [

--- a/src/tests/test_pyodide.py
+++ b/src/tests/test_pyodide.py
@@ -146,6 +146,11 @@ def test_monkeypatch_eval_code(selenium):
     )
     assert selenium.run("x = 99; 5") == [3, 5]
     assert selenium.run("7") == [99, 7]
+    selenium.run(
+        """
+        pyodide.eval_code = old_eval_code
+        """
+    )    
 
 
 def test_hiwire_is_promise(selenium):

--- a/src/tests/test_pyodide.py
+++ b/src/tests/test_pyodide.py
@@ -154,6 +154,7 @@ def test_monkeypatch_eval_code(selenium):
             """
         )
 
+
 def test_hiwire_is_promise(selenium):
     for s in [
         "0",

--- a/src/tests/test_pyodide.py
+++ b/src/tests/test_pyodide.py
@@ -150,7 +150,7 @@ def test_monkeypatch_eval_code(selenium):
         """
         pyodide.eval_code = old_eval_code
         """
-    )    
+    )
 
 
 def test_hiwire_is_promise(selenium):

